front/lib/api/assistant/actions/github.ts

```diff
@@ -0,0 +1,415 @@
+import logger from "@app/logger/logger";
+
+interface GithubGetPullRequestActionBlob {
+  id: ModelId;
+  agentMessageId: ModelId;
+  params: {
+    owner: string;
+    repo: string;
+    pullNumber: number;
+  };
+  pullBody: string | null;
+  pullCommits:
+    | {
+        sha: string;
+        author: string;
+        message: string;
+      }[]
+    | null;
+  pullDiff: string | null;
+  functionCallId: string | null;
+  functionCallName: string | null;
+  step: number;
+}
+
+export class GithubGetPullRequestAction extends BaseAction {
+  readonly agentMessageId: ModelId;
+  readonly params: {
+    owner: string;
+    repo: string;
+    pullNumber: number;
+  };
+  readonly pullBody: string | null;
+  readonly pullCommits:
+    | {
+        sha: string;
+        author: string;
+        message: string;
+      }[]
+    | null;
+  readonly pullDiff: string | null;
+  readonly functionCallId: string | null;
+  readonly functionCallName: string | null;
+  readonly step: number = -1;
+  readonly type = "github_get_pull_request_action";
+
+  constructor(blob: GithubGetPullRequestActionBlob) {
+    super(blob.id, "github_get_pull_request_action");
+    this.agentMessageId = blob.agentMessageId;
+    this.params = blob.params;
+    this.pullBody = blob.pullBody;
+    this.pullCommits = blob.pullCommits;
+    this.pullDiff = blob.pullDiff;
+    this.functionCallId = blob.functionCallId;
+    this.functionCallName = blob.functionCallName;
+    this.step = blob.step;
+  }
+
+  renderForFunctionCall(): FunctionCallType {
+    return {
+      id: this.functionCallId ?? `call_${this.id.toString()}`,
+      name:
+        this.functionCallName ?? DEFAULT_GITHUB_GET_PULL_REQUEST_ACTION_NAME,
+      arguments: JSON.stringify(this.params),
+    };
+  }
+
+  async renderForMultiActionsModel(): Promise<FunctionMessageTypeModel> {
+    const content = `${this.pullBody}\n\n${(this.pullCommits || [])
+      .map((c) => `${c.sha} ${c.author}: ${c.message}`)
+      .join("\n")}\n\n${this.pullDiff}`;
+
+    return {
+      role: "function" as const,
+      name:
+        this.functionCallName ?? DEFAULT_GITHUB_GET_PULL_REQUEST_ACTION_NAME,
+      function_call_id: this.functionCallId ?? `call_${this.id.toString()}`,
+      content,
+    };
+  }
+}
+
+/**
+ * Params generation.
+ */
+
+export class GithubGetPullRequestConfigurationServerRunner extends BaseActionConfigurationServerRunner<GithubGetPullRequestConfigurationType> {
+  // Generates the action specification for generation of rawInputs passed to `run`.
+  async buildSpecification(
+    _auth: Authenticator,
+    { name, description }: { name: string; description: string | null }
+  ): Promise<Result<AgentActionSpecification, Error>> {
+    return new Ok({
+      name,
+      description:
+        description ?? DEFAULT_GITHUB_GET_PULL_REQUEST_ACTION_DESCRIPTION,
+      inputs: [
+        {
+          name: "owner",
```

Comment:
What context enables the assistant to fill this value ? Same for repo

Reply:
A link shared directly or instructions